py/server/deephaven/table.py

```diff
@@ -1343,6 +1345,76 @@ def __init__(self, j_partitioned_table):
         self._constituent_changes_permitted = None
         self._is_refreshing = None
 
+    @classmethod
+    def from_partitioned_table(cls,
+                               table: Table,
+                               key_cols: Union[str, List[str]] = None,
+                               unique_keys: bool = None,
+                               constituent_column: str = None,
+                               constituent_table_columns: List[Column] = None,
+                               constituent_changes_permitted: bool = None) -> PartitionedTable:
+        """Creates a PartitionedTable from the provided underlying partitioned Table.
+
+        Note: key_cols, unique_keys, constituent_table_columns, constituent_table_columns,
+        constituent_changes_permitted must either be all None or all have values
+
+        Args:
+            table (Table): the underlying partitioned table
+            key_cols (Union[str, List[str]]): the key column name(s) of 'table'
+            unique_keys (bool): whether the keys in 'table' are guaranteed to be unique
+            constituent_column (str): the constituent column name in 'table'
+            constituent_table_columns (list[Column]): the column definitions of the constituent table
+            constituent_changes_permitted (bool): whether the values of the constituent column can change
+
+        Returns:
+            a PartitionedTable
+
+        Raise:
+            DHError
+        """
+        none_args = [key_cols, unique_keys, constituent_table_columns, constituent_table_columns,
+                     constituent_changes_permitted]
+
+        try:
+            if all([arg is None for arg in none_args]):
+                return PartitionedTable(j_partitioned_table=_JPartitionedTableFactory.of(table.j_table))
+
+            if all([arg is not None for arg in none_args]):
+                table_def = _JTableDefinition.of([col.j_column_definition for col in constituent_table_columns])
+                j_partitioned_table = _JPartitionedTableFactory.of(table.j_table, j_array_list(to_sequence(key_cols)),
```

Comment:
Formatting seems inconsistent on this line. Why put two args on this line, but not subsequent ones?